ui-tests/src/test/java/org/openmrs/reference/AddFormTest.java

```diff
@@ -2,59 +2,77 @@
 import org.openmrs.reference.page.HeaderPage;
 import org.openmrs.reference.page.HomePage;
+import org.openmrs.reference.page.HtmlFormsPage;
 import org.openmrs.reference.page.ManageFormsPage;
+import org.openmrs.reference.page.ManageHtmlFormsPage;
 import org.openmrs.uitestframework.test.TestBase;
+import org.openqa.selenium.By;
 
 import static org.junit.Assert.assertNotNull;
 
-
 /**
  * Created by nata on 24.06.15.
  */
 public class AddFormTest extends TestBase {
-    private HomePage homePage;
-    private HeaderPage headerPage;
-    private ManageFormsPage manageForm;
-    private ClinicianFacingPatientDashboardPage patientDashboardPage;
-
-    @Before
-    public void setUp() throws Exception {
-        homePage = new HomePage(page);
-        assertPage(homePage);
-        headerPage = new HeaderPage(driver);
-        manageForm = new ManageFormsPage(driver);
-        patientDashboardPage = new ClinicianFacingPatientDashboardPage(page);
-    }
+  private static String name = "newFormTest1";
+  private static String description = "description of new form";
+  private static String version = "1.2";
+  private HomePage homePage;
+  private HeaderPage headerPage;
+  private ManageFormsPage manageForm;
+  private AdministrationPage administrationPage;
+  private ClinicianFacingPatientDashboardPage patientDashboardPage;
+  private ManageHtmlFormsPage manageHtmlFormsPage;
+  private HtmlFormsPage htmlFormsPage;
 
-    @Ignore //ignore due to moving forms functionality
-    @Test
-    public void addFormTest() throws Exception {
-        homePage.goToManageForm();
-        if (!manageForm.addPresent()) {
-            manageForm.delete();
-        }
-        manageForm.add();
-        manageForm.addLabel("Eye Report");
-        manageForm.addIcon("icon-align-justify");
-        manageForm.formIdFromUrl();
-        manageForm.save();
-        headerPage.clickOnHomeIcon();
-        homePage.goToActiveVisitPatient();
-        assertNotNull("Eye Report", patientDashboardPage.FORM_EXIST);
-        headerPage.clickOnHomeIcon();
-        homePage.goToManageForm();
-        manageForm.deletePath();
+  @Before
+  public void setUp() throws Exception {
+	homePage = new HomePage(page);
+	administrationPage = new AdministrationPage(page);
+	headerPage = new HeaderPage(driver);
+	htmlFormsPage = new HtmlFormsPage(page);
+	manageHtmlFormsPage = new ManageHtmlFormsPage(page);
+	manageForm = new ManageFormsPage(driver);
+	patientDashboardPage = new ClinicianFacingPatientDashboardPage(page);
+  }
 
-    }
+  @Test
+  @Category(BuildTests.class)
+  public void addFormTest() throws Exception {
+	manageHtmlFormsPage = homePage.goToAdministration().clickOnManageHtmlForms();
+	if (manageHtmlFormsPage.getElementsIfExisting(By.xpath("//*[contains(text(), '" + name + "')]")).isEmpty()) {
+		manageHtmlFormsPage.clickOnNewHtmlForm();
+		htmlFormsPage.createNewFormTest(name, description, version);
+    }	
```

Comment:
this should be on the same line as next line, fix the above inside block

Reply:
resolved thanks

Reply:
but it looks wrong even now